inference-engine/src/mkldnn_plugin/nodes/mkldnn_topk_node.cpp

```diff
@@ -40,190 +1230,466 @@ bool MKLDNNTopKNode::isSupportedOperation(const std::shared_ptr<ngraph::Node>& o
+    config.outConfs[TOPK_INDEX].inPlace = -1;
 
-    addSupportedPrimDesc({{TensorDescCreatorTypes::ncsp, Precision::FP32},
-                          {TensorDescCreatorTypes::ncsp, Precision::I32}},
-                         outDataConf,
-                         impl_desc_type::ref_any);
+    auto pushDesc = [&](memory::format_tag inFormat, memory::format_tag outFormat, memory::data_type dataType, impl_desc_type impl_type) {
+        config.inConfs[TOPK_DATA].desc = MKLDNNMemoryDesc(getParentEdgeAt(TOPK_DATA)->getDims(), dataType, inFormat);
+        config.inConfs[TOPK_K].desc = MKLDNNMemoryDesc(getParentEdgeAt(TOPK_K)->getDims(), memory::data_type::s32, memory::format_tag::x);
+        config.outConfs[TOPK_DATA].desc = MKLDNNMemoryDesc(getChildEdgeAt(TOPK_DATA)->getDims(), dataType, outFormat);
+        config.outConfs[TOPK_INDEX].desc = MKLDNNMemoryDesc(getChildEdgeAt(TOPK_INDEX)->getDims(), memory::data_type::s32, outFormat);
+        supportedPrimitiveDescriptors.push_back({config, impl_type, outFormat});
+    };
+
+    if (jit_mode) {
+        impl_desc_type impl_type = impl_desc_type::jit_sse42;
+        if (mayiuse(cpu::x64::avx512_common)) {
+            impl_type = impl_desc_type::jit_avx512;
+        } else if (mayiuse(cpu::x64::avx2)) {
+            impl_type = impl_desc_type::jit_avx2;
+        }
+
+        pushDesc(MKLDNNMemory::GetPlainFormat(memory::dims(getParentEdgeAt(TOPK_DATA)->getDims().ndims())),
+             MKLDNNMemory::GetPlainFormat(memory::dims(getChildEdgeAt(TOPK_DATA)->getDims().ndims())), data_type, impl_type);
+        if (getParentEdgeAt(TOPK_DATA)->getDims().ndims() == 4) {
+            if (mayiuse(cpu::x64::avx512_common)) {
+                pushDesc(memory::format_tag::nhwc, memory::format_tag::nhwc, data_type, impl_type);
+                pushDesc(memory::format_tag::nChw16c, memory::format_tag::nChw16c, data_type, impl_type);
+            } else if (mayiuse(cpu::x64::avx2) || mayiuse(cpu::x64::sse41)) {
+                pushDesc(memory::format_tag::nhwc, memory::format_tag::nhwc, data_type, impl_type);
+                pushDesc(memory::format_tag::nChw8c, memory::format_tag::nChw8c, data_type, impl_type);
+            }
+        } else if (getParentEdgeAt(TOPK_DATA)->getDims().ndims() == 5) {
+            if (mayiuse(cpu::x64::avx512_common)) {
+                pushDesc(memory::format_tag::ndhwc, memory::format_tag::ndhwc, data_type, impl_type);
+                pushDesc(memory::format_tag::nCdhw16c, memory::format_tag::nCdhw16c, data_type, impl_type);
+            } else if (mayiuse(cpu::x64::avx2) || mayiuse(cpu::x64::sse41)) {
+                pushDesc(memory::format_tag::ndhwc, memory::format_tag::ndhwc, data_type, impl_type);
+                pushDesc(memory::format_tag::nCdhw8c, memory::format_tag::nCdhw8c, data_type, impl_type);
+            }
+        }
+    } else {
+        pushDesc(MKLDNNMemory::GetPlainFormat(memory::dims(getParentEdgeAt(TOPK_DATA)->getDims().ndims())),
```

Comment:
Have you tried to force reference implementation?
I was curious to check the **topk** reference implementation but actually this ref descriptor is filtered out inside InitDescriptors().
Could you please double check?

Reply:
In fact I had double checked about reference implementation. All the "ref" test cases in topk cpu single layout tests had passed, after I forced jit_mode to be false. Thansk Egor!